src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,42 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_AGE = 18;
+    private static final int MINIMUM_LOGIN_LENGTH = 6;
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age cannot be null!");
+        }
+        if (user.getAge() < MINIMUM_AGE) {
+            throw new RegistrationException("You have to be over " + MINIMUM_AGE + "!");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null!");
+        }
+        if (user.getLogin().length() < MINIMUM_LOGIN_LENGTH) {
+            throw new RegistrationException("Login is too short! Minimum length: "
+                    + MINIMUM_LOGIN_LENGTH);
```

Comment:
Consider using a consistent message format for exceptions. For instance, you could include the actual length of the login or password in the exception message to provide more context to the user.